node/src/components/consensus/highway_core/state/tests.rs

```diff
@@ -202,6 +202,35 @@ fn add_vote() -> Result<(), AddVoteError<TestContext>> {
     Ok(())
 }
 
+#[test]
+fn ban_and_mark_faulty() -> Result<(), AddVoteError<TestContext>> {
+    let mut rng = TestRng::new();
+    let params = Params::new(
+        0,
+        TEST_BLOCK_REWARD,
+        TEST_BLOCK_REWARD / 5,
+        4,
+        u64::MAX,
+        Timestamp::from(u64::MAX),
+    );
+    // Everyone already knows Alice is faulty, so she is banned.
+    let mut state = State::new(WEIGHTS, params, vec![ALICE]);
+
+    assert_eq!(panorama![F, N, N], *state.panorama());
+    let err = vote_err(add_vote!(state, rng, ALICE, 0xA; N, N, N).err().unwrap());
+    assert_eq!(VoteError::Banned, err);
+
+    state.mark_faulty(ALICE); // No change: Banned state is permanent.
+    assert_eq!(panorama![F, N, N], *state.panorama());
+    let err = vote_err(add_vote!(state, rng, ALICE, 0xA; N, N, N).err().unwrap());
+    assert_eq!(VoteError::Banned, err);
+
+    // Now we also received external evidence (i.e. not in this instance) that Bob is faulty.
+    state.mark_faulty(BOB);
+    assert_eq!(panorama![F, F, N], *state.panorama());
```

Comment:
Can you show that `BOB` is `Fault::Indirect` and that we will accept BOB's messages?